pkg/platform/runtime/analytics/analytics.go

```diff
@@ -0,0 +1,25 @@
+package analytics
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	anaConsts "github.com/ActiveState/cli/internal/analytics/constants"
+	"github.com/ActiveState/cli/internal/analytics/dimensions"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/osutils"
+)
+
+var isExecutor bool
+
+func init() {
+	if osutils.Executable() == constants.StateExecutorCmd {
+		isExecutor = true
+	}
```

Comment:
This won't work because we rename executors to eg. python.exe.

But more importantly, executors do not import the analytics package. Instead they contact the svc here:

https://github.com/ActiveState/cli/blob/7d126eb0eda866520ebfc5ce0e6ac8fe13aa03da/cmd/state-exec/main.go#L87-L87

Which is handled in the svc here:

https://github.com/ActiveState/cli/blob/7d126eb0eda866520ebfc5ce0e6ac8fe13aa03da/internal/svcctl/comm.go#L84-L84

So I think you can spoof the source here:

https://github.com/ActiveState/cli/blob/7d126eb0eda866520ebfc5ce0e6ac8fe13aa03da/internal/svcctl/comm.go#L141-L141

Reply:
Thanks for the note about executor names. I meant to use `runtime/executors.IsExecutor()` but it looks like that change got lost in the churn.

As for executors and analytics, I did make that change here: https://github.com/ActiveState/cli/pull/2750/files#diff-f69f873b04661f16e266f4298e50a2f0ee7040fc6810650aad57d7e08d407b96R141. Perhaps your IDE didn't show it.